packages/taquito-tzip16/src/viewKind/michelson-storage-view.ts

```diff
@@ -169,7 +170,26 @@ export class MichelsonStorageView implements View {
             balance: '0'
         };
 
-        const result: any = await this.rpc.runCode(viewScript as any);
+        let result: any;
+        try {
+          result = await this.rpc.runCode(viewScript as any);
+        } catch(error: any) {
+          const parsedError = JSON.parse(error.body);
```

Comment:
I am wondering if there is a risk of having an error body that is an invalid JSON, JSON.parse would result in an "Unexpected token.." error
I have a similar comment for the .forEach at line 179, I am wondering what will be the behavior if the parsed body is not an array.

I think that the `extractFailedWithDetails` function could be reused here. https://github.com/ecadlabs/taquito/blob/master/packages/taquito/src/contract/errors.ts#L66
If the body is valid JSON, and if it contains an array of errors, it extracts the "with" property of the last error which contains the information about the script execution failure.

Reply:
that's a very good point. `extractFailedWithDetails` seem to have all the checks we need, so I will be using that. Will amend